File: frontend/src/game/upgrades/curses/health_reduc_2.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const HealthReduc2Def: UpgradeDef = {
  id: "health_reduc_2",
  name: "Reduced Health 2",
  description: "-10 max health.",
  rarity: RarityID.Uncommon,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: -10,
  isMultiplier: false,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class HealthReduc2 extends Upgrade {}

File: backend/app/core/upgrades/curses/health_reduc_2.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class HealthReduc2(UpgradeImplementation):
    id = 'health_reduc_2'
    name = 'Reduced Health 2'
    description = '-10 max health.'
    rarity = 'uncommon'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -10
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True
